File: fullstack-demo/server/src/auth/auth.controller.ts
import { Controller, Post, Get, Body, HttpCode } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { Repository } from 'typeorm';
import { AuthService } from './auth.service';
import { User } from './user.entity';
import { RegisterDto } from './dto/register.dto';
import { LoginDto } from './dto/login.dto';

@Controller('auth')
export class AuthController {
  constructor(
    private readonly authService: AuthService,
    @InjectRepository(User) private readonly userRepo: Repository<User>,
  ) {}

  // GET /auth/users —— 调试：查看所有注册用户（生产环境记得加 Guard！）
  @Get('users')
  async listUsers() {
    const users = await this.userRepo.find();
    // 剔除 password
    const safe = users.map(({ password, ...rest }) => rest);
    return { code: 0, data: safe, message: 'success' };
  }

  // POST /auth/register —— 注册
  @Post('register')
  async register(@Body() dto: RegisterDto) {
    const user = await this.authService.register(dto);
    const { password, ...safeUser } = user;
    return { code: 0, data: safeUser, message: '注册成功' };
  }

  // POST /auth/login —— 登录
  @Post('login')
  @HttpCode(200)
  async login(@Body() dto: LoginDto) {
    const result = await this.authService.login(dto);
    return { code: 0, data: result, message: '登录成功' };
  }
}

File: fullstack-demo/server-java/src/main/java/com/fullstack/demo/auth/AuthController.java
package com.fullstack.demo.auth;

import com.fullstack.demo.common.ApiResponse;
import jakarta.validation.Valid;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.stream.Collectors;

/**
 * 认证控制器 —— 对应 NestJS 的 AuthController
 * 路由前缀: /auth
 */
@RestController
@RequestMapping("/auth")
public class AuthController {

    private final AuthService authService;
    private final UserRepository userRepository;

    public AuthController(AuthService authService, UserRepository userRepository) {
        this.authService = authService;
        this.userRepository = userRepository;
    }

    /**
     * GET /auth/users —— 调试：查看所有注册用户
     * 对应 NestJS listUsers()
     */
    @GetMapping("/users")
    public ApiResponse<List<AuthService.SafeUser>> listUsers() {
        List<AuthService.SafeUser> safe = userRepository.findAll().stream()
                .map(user -> {
                    AuthService.SafeUser s = new AuthService.SafeUser();
                    s.setId(user.getId());
                    s.setUsername(user.getUsername());
                    s.setEmail(user.getEmail());
                    s.setCreatedAt(user.getCreatedAt());
                    s.setUpdatedAt(user.getUpdatedAt());
                    return s;
                })
                .collect(Collectors.toList());
        return ApiResponse.success(safe);
    }

    /**
     * POST /auth/register —— 注册
     * 对应 NestJS register()
     */
    @PostMapping("/register")
    public ApiResponse<AuthService.SafeUser> register(@Valid @RequestBody RegisterDto dto) {
        User user = authService.register(dto);
        AuthService.SafeUser safe = new AuthService.SafeUser();
        safe.setId(user.getId());
        safe.setUsername(user.getUsername());
        safe.setEmail(user.getEmail());
        safe.setCreatedAt(user.getCreatedAt());
        safe.setUpdatedAt(user.getUpdatedAt());
        return ApiResponse.success(safe, "注册成功");
    }

    /**
     * POST /auth/login —— 登录
     * 对应 NestJS login()
     */
    @PostMapping("/login")
    public ApiResponse<AuthService.LoginResult> login(@Valid @RequestBody LoginDto dto) {
        AuthService.LoginResult result = authService.login(dto);
        return ApiResponse.success(result, "登录成功");
    }
}
